### PR TITLE
[otel-infrastructure-collector] Add CRD generation + MySQL preset

### PR DESCRIPTION
**Pull request by matej-g**
# Description

Resolves ES-29.

Adds new features necessary for MySQL preset to be added to `otel-infrastructure-collector`:
- Adds CRD generation, to be used with OpenTelemetry Operator
- Adds and configures MySQL preset
- Documents both features in detail
- Use our fork of OpenTelemetry chart to enable these features

# How Has This Been Tested?

Locally on a cluster.

# Checklist:
- [X] I have updated the relevant Helm chart(s) version(s)
- [X] I have updated the relevant component changelog(s)
- [ ] This change does not affect any particular component (e.g. it's CI or docs change)

### CODE REVIEW

**Comment by matej-g** [comment]
FYI to fully implement all missing improvements and metrics, we are still blocked on https://github.com/coralogix/opentelemetry-helm-charts/pull/16#issuecomment-1635636767.

File: otel-infrastructure-collector/k8s-helm/README.md [hunk 1]
```diff
@@ -86,6 +118,95 @@ This configuration is filtering out any event that has the field `reason` with o
...
+  - This error indicates that the database user you provided does not have the required privileges to collect metrics. Provide the `PROCESS` privilege to the user, e.g. by running query
+    `GRANT PROCESS ON *.* TO 'user'@'%'`
+
+### Example preset configuration for single instance
+
+```yaml
+  mysql:
+    metrics:
+      enabled: true 
+      instances:
```
Comment (oded-dd) [inline otel-infrastructure-collector/k8s-helm/README.md:180]: For one instance I also needed `labelSelectors`

Comment (matej-g) [reply otel-infrastructure-collector/k8s-helm/README.md:180]: I tried this but I did not need to configure it. Could be in the previous version of the upstream chart this was not configure properly, I don't remember honestly in detail the changes. But I did test it without selectors and it works as expected.

**Comment by matej-g** [comment]
Superseded by https://github.com/coralogix/telemetry-shippers/pull/272, closing this one

**Pull request closed without merging**